src/commands/yubihsm/keys/import.rs

```diff
@@ -0,0 +1,125 @@
+extern crate serde_json;
+
+use super::*;
+
+use std::fs::File;
+use std::io::prelude::*;
+use std::process;
+use std::str;
+
+use abscissa::Callable;
+use signatory::ed25519;
+use signatory::ed25519::Seed;
+use subtle_encoding::base64;
+use tendermint::public_keys::ConsensusKey;
+use yubihsm;
+
+use serde_json::Value;
+
+/// The `yubihsm keys import` subcommand
+#[derive(Debug, Default, Options)]
+pub struct ImportCommand {
+    /// Path to the validator configuration file
+    #[options(short = "p", long = "path")]
+    pub path: Option<String>,
+
+    /// Type of key to generate (default 'ed25519')
+    #[options(short = "t")]
+    pub key_type: Option<String>,
+
+    /// Label for imported key
+    #[options(short = "l", long = "label")]
+    pub label: Option<String>,
+
+    /// Key ID for imported key
+    #[options(free)]
+    key_id: Option<u16>,
+}
+
+impl Callable for ImportCommand {
+    fn call(&self) {
+        if self.path.is_none() {
+            status_err!("must provide a valid path to priv_validator.json");
+            process::exit(1);
+        }
+
+        if self.key_id.is_none() {
+            status_err!("must provide a unique key_id");
+            process::exit(1);
+        }
+
+        match &self.key_type {
+            Some(ref key_type) => if key_type != DEFAULT_KEY_TYPE {
+                status_err!(
+                    "only supported key type is: ed25519 (given: \"{}\")",
+                    key_type
+                );
+                process::exit(1);
+            },
+            None => (),
+        }
+
+        if let Some(path) = &self.path {
+            let mut f = File::open(path).unwrap_or_else(|e| {
+                status_err!("couldn't open validator config file {}: {}", path, e);
+                process::exit(1);
+            });
+
+            let mut contents = Vec::new();
+            f.read_to_end(&mut contents).unwrap_or_else(|e| {
+                status_err!("couldn't read validator config file {}: {}", path, e);
+                process::exit(1);
+            });
+            let v: Value = serde_json::from_slice(&contents).unwrap();
+            let s = v["priv_key"]["value"].as_str().unwrap_or_else(|| {
```

Comment:
One alternative to consider for this sort of thing is using `serde_derive` to derive the structure of `priv_validator.json`.

I'm not sure it matters as this code is probably effectively "done" in that the `priv_validator.json` format is unlikely to change and this is literally the only use case which matters for `tmkms`, but I thought I'd at least throw it out there.

Reply:
Thank! I also thought about this. I think we can keep this for now. If we ever need other fields, we can change this.